src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,46 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private int size;
+    private Node<T> head;
+
+    private static class Node<T> {
```

Comment:
maybe move this to the bottom of the class

Reply:
but I like it here...

Reply:
usually private elements are moved to the bottom of the class but it's not crucial